util/deephash/deephash.go

```diff
@@ -104,6 +104,70 @@ import (
 //	  by looking up the hash in a magical map that returns the set of entries
 //	  for that given hash.
 
+// SelfHasher is implemented by types that can compute their own hash
+// by writing values through the provided [Hasher] parameter.
+// Implementations must not leak the provided [Hasher].
+//
+// If the implementation of SelfHasher recursively calls [Hash],
+// then infinite recursion is quite likely to occur.
+// To avoid this, use a type definition to drop methods before calling [Hash]:
```

Comment:
```suggestion
// To avoid this, use the newtype pattern to provide an otherwise-identical type to `deephash.Hash`:
```

(Hash is ambiguous - in this case you mean deephash.Hash but on first read I thought you mean the interface method)

Reply:
Made it `[deephash.Hash]` instead to make it more explicit and also have it link properly in godoc.